Allow multiple pip entries in YAML file for `conda env create --file`

XuehaiPan opened the pull request:
The `conda env` dependency parser only keep the last entry for `pip` due to `dict.update()` at line 184:

https://github.com/conda/conda/blob/9e9461760bbd71a178220982e8da85637aa2f699/conda_env/env.py#L182-L186

`conda env create --file` with the following YAML file will only install `pip-pkg-Z` with `pip` and drops `pip-pkg-{X,Y}` when doing dependency parsing.

```yaml
dependencies:
  - conda-pkg-A
  - pip:
      - pip-pkg-X
      - pip-pkg-Y
  - conda-pkg-B
  - pip:
      - pip-pkg-Z
  - conda-pkg-C
```

If I want to install all `pip` packages, I should merge all `- pip:` entries into one in the YAML file, like:

```yaml
dependencies:
  - conda-pkg-A
  - pip:
      - pip-pkg-X
      - pip-pkg-Y
      - pip-pkg-Z
  - conda-pkg-B
  - conda-pkg-C
```

But I hope `conda env` could merge these automatically, and I can keep my YAML recipe in a more structured way. For example, I can put the relevant packages together:

```yaml
dependencies:
  # Some core dependencies
  - conda-pkg-A
  - pip:
      - pip-pkg-X
      - pip-pkg-Y
  - conda-pkg-B
  - conda-pkg-C

  # For documentation
  - sphinx
  - sphinx-autobuild
  - pip:
      - sphinxcontrib-tikz
```

jezdez commented:
Interesting idea, would you please mind adding a test case for this? There should be prior tests in this area in `tests/conda_env`. Thank you!

XuehaiPan commented:
I have updated the test cases for `Dependencies.parse` method.

jezdez commented:
@XuehaiPan Would you mind adding a news item for the changelog and update the docs to make sure users know about this new feature? Thank you!

XuehaiPan commented:
> Would you mind adding a news item for the changelog and update the docs to make sure users know about this new feature?

I have added the news item and updated the docs.

XuehaiPan commented on docs/source/user-guide/tasks/manage-environments.rst:
```diff
@@ -857,6 +857,32 @@ EXAMPLE: A more complex environment file:
    to get any bug fixes whilst still maintaining consistency of
    software environment.
 
+Starting from conda 4.14.0 release, multiple entries of ``pip:`` can be used for
```
The version number 4.14.0 here is the next version of conda release (currently 4.13.0). This is arbitrary taken in the document. It should be updated before we merge this PR.

XuehaiPan commented:
Gentle ping for this.
